### PR TITLE
Describe behavior of `copy_grants`

### PR DESCRIPTION
**Pull request by dbeatty10**
[Page preview](https://deploy-preview-3474--docs-getdbt-com.netlify.app/reference/resource-configs/snowflake-configs#copying-grants)

## What are you changing in this pull request and why?
See https://github.com/dbt-labs/dbt-snowflake/issues/567#issuecomment-1577441385 for context.

One page with edits:
- https://deploy-preview-3474--docs-getdbt-com.netlify.app/reference/resource-configs/snowflake-configs#copying-grants

One page that _could_ have edits, but doesn't currently:
- https://deploy-preview-3474--docs-getdbt-com.netlify.app/reference/resource-configs/grants#database-specific-requirements-and-notes

## Checklist
- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Don't merge until the bug is fixed and released within dbt-snowflake
- [ ] Verify that the versioning components align with the intended versions

### CODE REVIEW

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 1]
```diff
@@ -265,10 +265,27 @@ select * from index_sessions
...
+
+<VersionBlock firstVersion="1.2" lastVersion="1.5">
+
 When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. The default value is `false`.
 
+</VersionBlock>
+
+<VersionBlock firstVersion="1.6">
+
+When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. After materialization, it will also apply the `grants` that are [configured](/reference/resource-configs/grants#configuring-grants). No grants are revoked when `copy_grants` config is set to `true`.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/snowflake-configs.md:282]:
```suggestion
When you set the `copy_grants` config to `true`, dbt adds the `copy grants` <Term id="ddl" /> qualifier when it rebuilds tables and <Term id="view">views</Term>. After materialization, it also applies the `grants` that are [configured](/reference/resource-configs/grants#configuring-grants). When the `copy_grants `config is set to `true`, no grants are revoked.
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 2]
```diff
@@ -265,10 +265,27 @@ select * from index_sessions
...
+
+When the `copy_grants` config is set to `false`, dbt will revoke any existing grants prior to applying the configured `grants`.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/snowflake-configs.md:284]:
```suggestion
When you set the `copy_grants` config to `false`, dbt revokes any existing grants before applying the configured `grants`.
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 3]
```diff
@@ -265,10 +265,27 @@ select * from index_sessions
...
+
+The default value is `false`.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/snowflake-configs.md:286]:
```suggestion
The default value for the `copy_grants` config is set to `false`.
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 4]
```diff
@@ -265,10 +265,27 @@ select * from index_sessions
...
 
+<VersionBlock firstVersion="1.2">
+
 ## Copying grants
 
+</VersionBlock>
+
+<VersionBlock firstVersion="1.2" lastVersion="1.5">
+
 When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. The default value is `false`.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. The default value for the `copy_grants` config is set to `false`.
```

**Review by mirnawong1 — APPROVED** [review]
hey @dbeatty10 , left some non blocking suggestions to switch most of the content to active voice. please feel free to take them or not but you're good to merge either way!

deploy preview looks good!

**Comment by dbeatty10** [comment]
Thank you @mirnawong1 ! I adopted all of your suggestions 🧠 

We will wait until https://github.com/dbt-labs/dbt-snowflake/issues/567 is resolved to do the final merge on this.

**Comment by matthewshaver** [comment]
Hi @dbeatty10 - Is this draft still relevant and do you know when we will need to address it? The Snowflake PR from which this originates is still open, but the version block text indicates this was (or should have been) a v1.6 update.

**Comment by dbeatty10** [comment]
> Hi @dbeatty10 - Is this draft still relevant and do you know when we will need to address it? The Snowflake PR from which this originates is still open, but the version block text indicates this was (or should have been) a v1.6 update.

It's actually not 100% clear to me what the original intent for this feature was vs. how this config behaves currently.

So the main blocker to merging this is that I'm actually not confident the changes are _true_. I'm also not sure if the currently documentation is accurate. Also not sure if there's a bug in the current implementation or not 😅.

**Comment by matthewshaver** [comment]
Closing aging drafts. Given the originating issue is still open, we can re-open or otherwise address this once those changes happen. If it's preferable to address earlier, feel free to re-open this at any time.

**Pull request closed without merging**